docs/fassets/developer-guides/12-fassets-autoredeem.mdx

```diff
@@ -0,0 +1,580 @@
+   DEPLOYER_PRIVATE_KEY=your_private_key_here
+   COSTON2_COMPOSER=0x... # Required! Set this after deploying the composer
+   HYPERLIQUID_FXRP_OFT=0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+   ```
+
+3. **Update XRP Address**:
+   - Edit `CONFIG.XRP_ADDRESS` in the script to your XRP ledger address
+   - This is where you'll receive the native XRP
+   - Must be a valid XRP address format (starts with 'r')
+
+4. **Run the Script on Hyperliquid Testnet**:
+   ```bash
+   yarn hardhat run scripts/fassets/autoRedeemHyperEVM.ts --network hyperliquidTestnet
+   ```
+
+### Expected Output
+
+```
+Using account: 0x742d35Cc6634C0532925a3b844Bc454e4438f44e
+✓ Composer configured: 0x123...
+
+📋 Redemption Parameters:
+Amount: 10.0 FXRP
+XRP Address: rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp
+Redeemer: 0x742d35Cc6634C0532925a3b844Bc454e4438f44e
+
+Connecting to FXRP OFT on Hyperliquid EVM: 0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+
+✓ Connected to FXRP OFT: 0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+OFT address: 0x14bfb521e318fc3d5e92A8462C65079BC7d4284c
+
+Compose message encoded
+
+💰 Current FXRP balance: 25.0
+Sufficient balance
+
+💵 LayerZero Fee: 0.002456 HYPE
+
+🚀 Sending 10.0 FXRP to Coston2 with auto-redeem...
+Target composer: 0x123...
+Underlying address: rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp
+
+✓ Transaction sent: 0xdef456...
+Waiting for confirmation...
+✅ Confirmed in block: 9876543
+
+🎉 Success! Your FXRP is on the way to Coston2!
+
+📊 Track your cross-chain transaction:
+https://testnet.layerzeroscan.com/tx/0xdef456...
+
+⏳ The auto-redeem will execute once the message arrives on Coston2.
+XRP will be sent to: rpHuw4bKSjonKRrKKVYUZYYVedg1jyPrmp
+```
+
+<details>
+<summary>View `autoRedeemHyperEVM.ts` source code</summary>
+
+<CodeBlock language="typescript" title="scripts/fassets/autoRedeemHyperEVM.ts">
+  {autoRedeemHyperEVM}
+</CodeBlock>
+
+</details>
+
+## FAQ
+
+**Q: How long does bridging take?**
+A: Typically 2-5 minutes for LayerZero message delivery + execution time.
+
+**Q: What's the minimum amount I can bridge?**
+A: Any amount for bridging to Hyperliquid. For auto-redeem, minimum is 1 [lot](https://dev.flare.network/fassets/minting#lots) (10 FXRP for XRP).
+
+**Q: Can I bridge to a different address?**
+A: Yes, edit the `recipientAddress` parameter in the scripts.
+
+**Q: What happens if compose execution fails?**
+A: Tokens will be stuck in the composer. Owner can recover using `recoverTokens()`.
+
+**Q: Can I use this on mainnet?**
+A: This is designed for testnet. For mainnet, update contract addresses, thoroughly test, and audit all code.
+
+**Q: How do I get FTestXRP on Flare Testnet Coston2?**
+A: Use Flare's FAsset minting process via the AssetManager contract.
+
+**Q: What if I don't have HYPE tokens?**
+A: Get them from Hyperliquid testnet [faucet](https://hyperliquid.gitbook.io/hyperliquid-docs/onboarding/testnet-faucet) or DEX.
+
+## Discovering Available Bridge Routes
```

Comment:
Now I'm thinking that this can be additional guide not in this one because it is already long.